src/server.js

```diff
@@ -1,38 +1,40 @@
-const app = require("./app");
-const mongoose = require("mongoose");
+const mongoose = require('mongoose');
+const server = require('./app').server;
+const socketListener = require('./socket');
 
-const { PORT, DATABASE_URL } = require("./config");
+const { PORT, DATABASE_URL } = require('./config');
 
 mongoose.Promise = global.Promise;
 
-let server;
-
 function runServer(databaseUrl, port = PORT) {
   return new Promise((resolve, reject) => {
     // mongoose.set('debug', true);
-    mongoose.set("useCreateIndex", true);
-    
-    mongoose.connect(databaseUrl, { useNewUrlParser: true,useUnifiedTopology: true}, err => {
+    mongoose.set('useCreateIndex', true);
+
+    mongoose.connect(databaseUrl, { useNewUrlParser: true, useUnifiedTopology: true }, err => {
       if (err) {
         return reject(err);
       }
-      server = app
-        .listen(port, () => {
+
+      server
+        .listen((port = PORT), () => {
```

Comment:
Could you explain this change? We are specifying port on line number 9 in the argument.

Reply:
Thanks for spotting this. The change is unnecessary.